File: chat-room-common/src/main/java/io/github/wj0410/chatroom/common/message/NormalMessage.java
package io.github.wj0410.chatroom.common.message;

import io.github.wj0410.chatroom.common.enums.ChatType;
import io.github.wj0410.chatroom.common.enums.old.MessageContainerType;
import io.github.wj0410.chatroom.common.model.MessageContainer;
import lombok.Data;
import org.springframework.util.CollectionUtils;

import java.util.List;

/**
 * 普通消息
 *
 * @author wangjie
 * @date 2023/10/25
 */
@Data
public class NormalMessage {
    private ChatType chatType;
    private String fromAccount;
    private String fromNickName;
    private String fromClientId;
    private String fromClientAvatar;
    private List<String> targetClientIds;
    private long timestamp;
    private List<MessageContainer> msgList;

    public NormalMessage() {
        this.timestamp = System.currentTimeMillis();
    }

    @Override
    public String toString() {
        String msg = "";
        if(!CollectionUtils.isEmpty(this.msgList)){
            for (MessageContainer messageContainer : this.msgList) {
                if (messageContainer.getType().equals(MessageContainerType.IMAGE)) {
                    msg += "[img],";
                } else {
                    msg += messageContainer.getText() + ",";
                }
            }
        }
        return "NormalMessage{" +
                "chatType=" + chatType +
                ", fromAccount='" + fromAccount + '\'' +
                ", fromNickName='" + fromNickName + '\'' +
                ", fromClientId='" + fromClientId + '\'' +
                ", fromClientAvatar=" + fromClientAvatar +
                ", targetClientIds=" + targetClientIds +
                ", timestamp=" + timestamp +
                ", msgList=" + String.format("[%s]", msg.replaceAll("^,|,$", "")) +
                '}';
    }
}

File: chat_vue/src/model/NormalMessage.ts
import {Message} from "./Message";
import {MessageType,ChatType} from "../constant/Enum";
import {MessageContainer} from "./MessageContainer";

export class NormalMessage {
    chatType: ChatType
    fromAccount: string
    fromNickName: string
    fromClientId: string
    fromClientAvatar: string
    targetClientIds: Array<string>
    timestamp: bigint
    msgList: Array<MessageContainer>

    constructor(data) {
        this.chatType = data.chatType;
        this.fromAccount = data.fromAccount;
        this.fromNickName = data.fromNickName;
        this.fromClientId = data.fromClientId;
        this.fromClientAvatar = data.fromClientAvatar;
        this.targetClientIds = data.targetClientIds;
        this.timestamp = data.timestamp;
        this.msgList = data.msgList;
    }

    buildMessage() {
        let obj = {
            type: MessageType.NORMAL,
            data: this
        }
        return new Message(obj);
    }
}
